shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -250,19 +250,38 @@ Engine::Engine(Delegate& delegate,
   // Setup the callback that will instantiate the rasterizer.
   flutter::Shell::CreateCallback<flutter::Rasterizer> on_create_rasterizer;
 #if defined(LEGACY_FUCHSIA_EMBEDDER)
-  on_create_rasterizer = [this](flutter::Shell& shell) {
+  on_create_rasterizer = [this, &product_config](flutter::Shell& shell) {
     if (use_legacy_renderer_) {
       FML_DCHECK(session_connection_);
       FML_DCHECK(surface_producer_);
       FML_DCHECK(legacy_external_view_embedder_);
 
+      if (product_config.enable_shader_warmup()) {
```

Comment:
Move this outside of the "if (use_legacy_renderer_)" block to avoid duplication )otherwise with my next comment you will have 3 of these)

Reply:
I sent some chats about this but basically the call to WarmupSkps() has to happen before the return statement which is different in the two branches of the if (use_legacy_renderer_) statement. so if i move this outside the if (use_legacy_renderer_) i will have to add another if (use_legacy_renderer_) following it to encapsulate the return statement. Personally i dont think this is an improvement but both are not great so if you prefer the alternative I'm happy to make that happen.

Reply:
Ahh I see.  Disregard then